lib/PriorityQueue.js

```diff
@@ -9,6 +9,7 @@ const Queue = require("./Queue");
 class PriorityQueue {
   constructor(size) {
     this._size = Math.max(+size | 0, 1);
+    /** @type {Queue[]} */
```

Comment:
This solved a lot error messages in this file.

(TS can guess many things with small annotations here and there)